overlord/state/change.go

```diff
@@ -149,6 +149,7 @@ type Change struct {
 	taskIDs            []string
 	ready              chan struct{}
 	lastObservedStatus Status
+	lastNoticedStatus  Status
```

Comment:
Nitpick: I don't love `lastNoticedStatus` since "noticed" in this case is a bit of a (clever) double-entendre. I had to check to be sure "noticed" referred to "recorded notice" rather than "noticed" as in "observed". Maybe `lastRecordedNoticeStatus` would be a bit clearer, even if it is more verbose?